tests/test_proxy_connect.py

```diff
@@ -17,31 +18,36 @@
 
 class HTTPSProxy(controller.Master, Thread):
 
-    def __init__(self, port):
+    def __init__(self):
         password_manager = http_auth.PassManSingleUser('scrapy', 'scrapy')
         authenticator = http_auth.BasicProxyAuth(password_manager, "mitmproxy")
         cert_path = os.path.join(os.path.abspath(os.path.dirname(__file__)),
             'keys', 'mitmproxy-ca.pem')
         server = proxy.ProxyServer(proxy.ProxyConfig(
             authenticator = authenticator,
             cacert = cert_path),
-            port)
+            0)
+        self.server = server
         Thread.__init__(self)
         controller.Master.__init__(self, server)
 
+    def http_address(self):
+        return 'http://%s:%d' % self.server.socket.getsockname()
+
 
 class ProxyConnectTestCase(TestCase):
 
     def setUp(self):
         self.mockserver = MockServer()
         self.mockserver.__enter__()
         self._oldenv = os.environ.copy()
-        self._proxy = HTTPSProxy(8888)
+
+        self._proxy = HTTPSProxy()
         self._proxy.start()
+
         # Wait for the proxy to start.
         time.sleep(1.0)
-        os.environ['http_proxy'] = 'http://scrapy:scrapy@localhost:8888'
```

Comment:
what's the reason http_proxy env variable is deleted?

Reply:
No reason, just a mistake